qiskit_machine_learning/neural_networks/sampling_neural_network.py

```diff
@@ -16,7 +16,18 @@
 from typing import Tuple, Union, List, Optional
 
 import numpy as np
-from sparse import SparseArray
+
+try:
+    from sparse import SparseArray
+except ImportError:
+
+    class SparseArray:  # type: ignore
+        """Empty SparseArray class
+        Replacement if sparse.SparseArray is not present.
+        """
+
+        pass
```

Comment:
This code is present in a few scripts. Any chance we can re-use it somehow? Or keep just one fake definition of `SparseArray`?

Reply:
The import and the "fake" SparseArray definitions need to be in the same file though. The "fake" SparseArray class is just because of the typehint definition not throwing errors if sparse is not installed and can only be declared if the real SparseArray is not present.

Reply:
What if 
```python
try:
    from sparse import SparseArray
except ImportError:
    from .utils import SparseArray    # there's a fake definition somewhere in qml
```
Or this does not save much as well?

Reply:
And then be careful that nobody ever imports utils by mistake if sparse exists. I believe it should not be called utils right ? As it might contain other things and if it contained the fake definition of SparseArray would prevent people from using it unless sparse is not installed. Maybe have a fake_sparse.py file with just this empty class definition and nothing else but then you just created an indirection and had to add a comment to explain what it meant for something that I could see immediately in the first case.

Reply:
Importing by a mistake is not great at all.